File: 1391-counting-elements/1391-counting-elements.java
class Solution {
    public int countElements(int[] arr) {
        Set<Integer> st = new HashSet<>();
        int res = 0;
        for(int n: arr){
            st.add(n);
        }
        for(int n: arr){
            if(st.contains(n + 1))
                res++;
        }
        return res;
    }
}

File: 1391-counting-elements/1391-counting-elements.rs
use std::collections::HashSet;
impl Solution {
    pub fn count_elements(arr: Vec<i32>) -> i32 {
        let mut set: HashSet<i32> = HashSet::new();
        let mut count = 0;
        for i in 0..arr.len(){
            set.insert((arr[i]));
        }
        for i in 0..arr.len(){
            if set.contains(&(arr[i] + 1)){
                count+= 1;
            }
        }
        count
    }
}